midend/interpreter.cpp

```diff
@@ -310,20 +310,35 @@ void SymbolicStruct::dbprint(std::ostream& out) const {
 }
 
 SymbolicHeaderUnion::SymbolicHeaderUnion(const IR::Type_HeaderUnion* type,
-                               bool uninitialized,
-                               const SymbolicValueFactory* factory) :
+                                         bool uninitialized,
+                                         const SymbolicValueFactory* factory) :
         SymbolicStruct(type, uninitialized, factory),
-        valid(new SymbolicBool(false)) {}
+        valid(new SymbolicBool(false)) {
+            auto fieldsClone = fieldValue;
+            bool validityFlag = false;
+            for (auto f : type->to<IR::Type_StructLike>()->fields) {
+                if (fieldsClone[f->name.name]->to<SymbolicHeader>()->valid->value) {
+                    validityFlag = true;
+                    break;
+                }
+            }
+            valid = new SymbolicBool(validityFlag);
+}
 
-void SymbolicHeaderUnion::setValid(bool v) {
-    if (!v)
-        setAllUnknown();
+void SymbolicHeaderUnion::setValid(bool v, cstring field) {
+    if (field) {
+        fieldValue[field]->to<SymbolicHeader>()->setValid(v);
+        for (auto f : type->to<IR::Type_StructLike>()->fields) {
+            if (f->name.name != field)
+                fieldValue[f->name.name]->setAllUnknown();
+        }
+    }
     valid = new SymbolicBool(v);
```

Comment:
Should a Header Union have a valid field? Considering it is dependent on its members.

Reply:
I'm not sure about this, I think it should be